Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -1356,6 +1356,16 @@ LayoutUnit RenderBlockFlow::collapseMarginsWithChildInfo(RenderBox* child, Rende
     return logicalTop;
 }
 
+bool RenderBlockFlow::shouldTrimChildMargin(MarginTrimType marginTrimType, const RenderBox& child) const
+{
+    auto blockChild = dynamicDowncast<RenderBlock>(child);
```

Comment:
how does this work with block-level replaced boxes?

Reply:
Ah I guess it would not work in that case. Perhaps instead I should try checking the display type of the child here

Reply:
The "Block" in RenderBlock is surely misleading and makes you think all block-level boxes end up being RenderBlocks (and that's one of the reasons why LFC ditched strong typing in layout tree). It's safer to consult the display type when  looking for blocks, inlines etc.

Reply:
Changed this line to use `child.style().isDisplayBlockLevel() `